src/app/components/GoogleSheetsInput.svelte

```diff
@@ -0,0 +1,172 @@
+   *  listeners.
+   */
+  function initClient() {
+    error = null;
+    gapi.client
+      .init({
+        apiKey: API_KEY,
+        clientId: CLIENT_ID,
+        discoveryDocs: DISCOVERY_DOCS,
+        scope: SCOPES,
+      })
+      .then(
+        () => {
+          // Listen for sign-in state changes.
+          gapi.auth2.getAuthInstance().isSignedIn.listen(updateSigninStatus);
+
+          // Handle the initial sign-in state.
+          updateSigninStatus(gapi.auth2.getAuthInstance().isSignedIn.get());
+        },
+        function () {
+          error = "Error: Could not connect to Google Sheets";
+        }
+      );
+  }
+
+  /**
+   *  Called when the signed in status changes, to update the UI
+   *  appropriately. After a sign-in, the API is called.
+   */
+  function updateSigninStatus(isSignedIn) {
+    if (isSignedIn) {
+      getValuesFromSpreadSheet();
+    } else {
+      handleAuthClick();
+      getValuesFromSpreadSheet();
+    }
+  }
+
+  /**
+   *  Sign in the user upon button click.
+   */
+  function handleAuthClick(event) {
+    gapi.auth2.getAuthInstance().signIn();
+  }
+
+  /**
+   *  Sign out the user upon button click.
+   */
+  function handleSignoutClick(event) {
+    gapi.auth2.getAuthInstance().signOut();
+  }
+
+  /**
+   * Gets the unique spreadsheet ID from a Google Sheet URL
+   *
+   */
+  function getSpreadsheetId() {
+    googleSheetsURL = document.getElementById("googleSheetsURL").value;
+    if (googleSheetsURL) {
+      return googleSheetsURL.split("/")[5];
+    } else {
+      error = `Error: The url ${googleSheetsURL} is not a valid Google Sheets URl. Please paste a valid one.`;
+    }
+  }
+
+  /**
+   * Read word and count form storage
+   * Sample: https://docs.google.com/spreadsheets/d/1lzHxoMWHpdGecby4d0y15AVf80csA7iNkaMMEL54Q7g/edit#gid=0
+   */
+  async function getValuesFromSpreadSheet() {
+    const wordListObject = [];
+    let spreadsheetId = getSpreadsheetId() || "";
+
+    if (error) {
+      return error;
+    }
+
+    gapi.client.sheets.spreadsheets.values
+      .get({
+        spreadsheetId: spreadsheetId,
+        range: "A2:B",
+      })
+      .then(
+        async function (response) {
+          var range = response.result;
+          if (range.values.length > 0) {
+            for (let i = 0; i < range.values.length; i++) {
+              let row = range.values[i];
+              let word = row[0];
+              let wordCount = row[1];
+              if (!wordCount) {
+                wordCount = 0;
+              }
+              wordListObject.push([word, wordCount]);
+            }
+            error = null;
+
+            await worker.readGoogleSheet(spreadsheetId, wordListObject);
+          } else {
+            error = "Error: No data found in the Google Sheet.";
+          }
+        },
+        function (response) {
+          error = "Error: " + response.result.error.message;
+        }
+      );
+  }
```

Comment:
If you declare a function `async`, do not mix both `async/await` syntax with Promise callbacks. Instead, use `async`/`await` throughout:

```suggestion

  async function getValuesFromSpreadSheet() {
    const wordListObject = [];
    const spreadsheetId = getSpreadsheetId() || "";
    if (error) {
      return error;
    }

    let response;
    try {
      response = await gapi.client.sheets.spreadsheets.values.get({
        spreadsheetId: spreadsheetId,
        range: "A2:B",
      });
    } catch {
      error = "Error: " + response.result.error.message;
      return;
    }

    const range = response.result;
    if (range.values.length <= 0) {
      error = "Error: No data found in the Google Sheet.";
      return;
    }

    for (let i = 0; i < range.values.length; i++) {
      const row = range.values[i];
      const word = row[0];
      let wordCount = row[1];
      if (!wordCount) {
        wordCount = 0;
      }
      wordListObject.push([word, wordCount]);
    }
    error = null;

    await worker.readGoogleSheet(spreadsheetId, wordListObject);
  }
```